src/ast/passes/codegen_llvm.cpp

```diff
@@ -1307,8 +1308,11 @@ void CodegenLLVM::visit(Call &call)
   } else if (call.func == "strerror") {
     auto scoped_del = accept(call.vargs.front());
   } else if (call.func == "strncmp") {
+    // Checked in semantic analysis
```

Comment:
This isn't on the hot path, I almost feel like the extra optional check is better than this comment. Also it protects if, for some reason, the semantic analysis check gets removed.

Reply:
It's not done this way for performance. The reason is b/c codegen is not responsible for error detection - that occurs in semantic analysis.

Reply:
So why not something like:
```
auto int_literal = bpftrace_.get_int_literal(call.vargs.at(2));
if (!int_literal) {
	LOG(BUG) << "Int literal should have been checked in semantic_analyser";
}
```

Reply:
Ah yeah that's better